gateway-js/src/__tests__/integration/configuration.test.ts

```diff
@@ -204,6 +204,67 @@ describe('gateway config / env behavior', () => {
     }
   });
 
+  describe('introspection headers', () => {
+    test('should allow not passing introspectionHeaders', async () => {
+      const requestCapture = jest.fn();
+      mockSDLQuerySuccess(service, requestCapture);
+
+      gateway = new ApolloGateway({
+        serviceList: [{ name: 'accounts', url: service.url }],
+      });
+
+      await gateway.load(mockApolloConfig);
+
+      expect(requestCapture.mock.calls[0][0]).toMatchObject({
+        headers: {
+          host: 'localhost:4001',
+        },
+      });
```

Comment:
> Note: in the case you want to match on nested partial objects, you need to wrap each partial with the expect.objectContaining.

Nice! Was not aware of that function!

Reply:
I couldn't get this to work, I guess because the `req` argument here is not a plain object, but an instance of `OverriddenClientRequest : http.ClientRequest`? Solved it like this instead:

```ts
const receivedHeaders = jest.fn()
nock.on("request", (req) => receivedHeaders(req.getHeaders()))
```